### PR TITLE
Fixed deathstab visual & fps bugs

### PR DESCRIPTION
**Pull request by nitoygo**
* Fixed FPS drop when casting deathstab
* Fixed incorrect deathstab animation

### CODE REVIEW

File: Source Main 5.2/source/w_ObjectInfo.h [hunk 1]
```diff
@@ -141,7 +141,7 @@ class OBJECT
     BYTE          Kind;
     WORD		Skill;
     BYTE		  m_byNumCloth;
-    BYTE		  m_byHurtByOneToOne;
+    BYTE		  m_byHurtByDeathstab;
```
Comment (sven-n) [inline Source Main 5.2/source/w_ObjectInfo.h]: float?

File: Source Main 5.2/source/ZzzCharacter.cpp [hunk 2]
```diff
@@ -3817,7 +3817,7 @@ void MoveCharacter(CHARACTER* c, OBJECT* o)
                 }
             }
         }
-        o->m_byHurtByOneToOne--;
+        o->m_byHurtByDeathstab--;
```
Comment (sven-n) [inline Source Main 5.2/source/ZzzCharacter.cpp]: -= FPS_ANIMATION_FACTOR ?

**Review by sven-n — COMMENTED** [review]
Thanks again. What do you think... shouldn't the `m_byHurtByDeathstab` be a float now? Because previously, it counted down per frame, now it should consider the fps

File: Source Main 5.2/source/w_ObjectInfo.h (hunk 1 above)
Comment (nitoygo) [reply Source Main 5.2/source/w_ObjectInfo.h]: i haven't thought of that. let me check...

Comment (nitoygo) [reply Source Main 5.2/source/w_ObjectInfo.h]: This is the part where it's set,  
`to->m_byHurtByDeathstab = 35;` 
Does this mean it's meant to stay for 35 frames? but now it should stay for the corresponding animation factor?

File: Source Main 5.2/source/GM_Raklion.cpp [hunk 3]
```diff
@@ -1905,7 +1905,7 @@ bool CGM_Raklion::RenderMonsterVisual(CHARACTER* c, OBJECT* o, BMD* b)
                         OBJECT* to = &tc->Object;
                         if (10 <= c->AttackTime && to->Live)
                         {
-                            to->m_byHurtByOneToOne = 35;
+                            to->m_byHurtByDeathstab = 35;
```
Comment (nitoygo) [inline Source Main 5.2/source/GM_Raklion.cpp:1908]: hey @sven-n, do we also need to update this initial value based on the fps animation factor? or is this fine, and we only need to update the part where it's decremented?

File: Source Main 5.2/source/w_ObjectInfo.h (hunk 1 above)
Comment (nitoygo) [reply Source Main 5.2/source/w_ObjectInfo.h]: I applied the suggestion, animation and duration seems fine.

File: Source Main 5.2/source/ZzzEffectJoint.cpp [hunk 4]
```diff
@@ -4235,7 +4235,7 @@ void MoveJoint(JOINT* o, int iIndex)
 
             vec3_t MagicPos;
             GetMagicScrew(iIndex * 17721, MagicPos, 1.4f);
-            VectorScale(MagicPos, 300.0f, MagicPos);
+            //VectorScale(MagicPos, 300.0f, MagicPos);
```
Comment (nitoygo) [inline Source Main 5.2/source/ZzzEffectJoint.cpp:4238]: Removing this line fixed the weird deathstab magic animation

**Pull request merged**